### PR TITLE
add a generic mesh shader and

### PR DESCRIPTION
**Pull request by headupinclouds**
and update hunter + huntergate

### CODE REVIEW

File: ogles_gpgpu/common/proc/mesh.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+);
+// clang-format on
+
+MeshShaderProc::MeshShaderProc(const VertexBuffer& vertices, const CoordBuffer& coords)
+    : texUnit(1)
+    , texTarget(GL_TEXTURE_2D)
+    , vertices(vertices)
+    , coords(coords)
+{
+    memset(MVP.data, 0, sizeof(MVP.data));
```
Comment (ruslo) [inline ogles_gpgpu/common/proc/mesh.cpp]: On Linux:
```
error: ‘memset’ was not declared in this scope
     memset(MVP.data, 0, sizeof(MVP.data));
```
I think `#include <string.h>` will fix it:
* https://linux.die.net/man/3/memset

File: ogles_gpgpu/ut/test-ogles_gpgpu.cpp [hunk 2]
```diff
@@ -721,6 +728,85 @@ TEST(OGLESGPGPUTest, TransformProc) {
...
+        mesh.setModelViewProjection(matrix);
+
+        video({ test.cols, test.rows }, test.ptr<void>(), true, 0, TEXTURE_FORMAT);
+
+        cv::Mat result;
+        getImage(mesh, result);
+        ASSERT_FALSE(result.empty());
+        
+        auto error = cv::norm(result, test);
+        ASSERT_LE(error, 1);
```
Comment (ruslo) [inline ogles_gpgpu/ut/test-ogles_gpgpu.cpp]: Test failed on Android emulator:
```
Expected: (error) <= (1), actual: 67.3127 vs 1
```

Comment (headupinclouds) [reply ogles_gpgpu/ut/test-ogles_gpgpu.cpp]: The tests pass locally running on the Android emulator.  I'll push some verbose test output to help troubleshoot.

File: ogles_gpgpu/common/proc/mesh.cpp (hunk 1 above)
Comment (headupinclouds) [reply ogles_gpgpu/common/proc/mesh.cpp]: Replaced with for loop initialization.

**Pull request merged**